src/algebra/group/pi.lean

```diff
@@ -235,6 +236,21 @@ lemma pi.single_mul [Π i, mul_zero_class $ f i] (i : I) (x y : f i) :
   single i (x * y) = single i x * single i y :=
 (mul_hom.single f i).map_mul x y
 
+@[to_additive]
+lemma pi.mul_single_commute [Π i, mul_one_class $ f i] :
+  pairwise (λ i j, ∀ (x : f i) (y : f j), commute (mul_single i x) (mul_single j y)) :=
+begin
+  intros i j hij x y, ext k,
+  by_cases h1 : i = k, { subst h1, simp [hij], },
+  by_cases h2 : j = k, { subst h2, simp [hij], },
+  simp [h1,  h2],
+end
+
+@[to_additive]
+lemma monoid_hom.single_commute [Π i, mul_one_class $ f i] :
```

Comment:
I'd argue there's no point having this one, since `monoid_hom.single_apply` can fire first through `simp`. It's similar to how we tend not to state injectivity of `submonoid.subtype`, since that gets simplified away too

Reply:
But yes, in this case I don’t need this lemma and can easily use the other one. Removed.